media/engine/webrtc_voice_engine.cc

```diff
@@ -2582,4 +2592,29 @@ void WebRtcVoiceMediaChannel::ConfigureEncoders(const webrtc::AudioEncoder::Conf
   }
 }
 
+void WebRtcVoiceMediaChannel::GetAudioLevels(
+    cricket::AudioLevel* captured_out,
+    cricket::ReceivedAudioLevel* received_out,
+    size_t received_out_size,
+    size_t* received_size_out) {
+
+  cricket::AudioLevel captured = 0;
+  for (const auto& kv : send_streams_) {
+    captured = kv.second->GetAudioLevel();
+  }
+  RTC_LOG(LS_INFO) << "FLUFF %%% captured: " << captured;
+
+  size_t received_size = 0;
+  for (const auto& kv : recv_streams_) {
+    received_out[received_size++] = cricket::ReceivedAudioLevel {
```

Comment:
Should received_out_size to be used for defense/validity?

Reply:
Yes, done